src/transformers/models/fuyu/image_processing_fuyu.py

```diff
@@ -468,22 +465,22 @@ def preprocess(
             # We assume that all images have the same channel dimension format.
             input_data_format = infer_channel_dimension_format(batch_images[0][0])
 
-        original_image_sizes = [get_image_size(images[0], channel_dim=input_data_format) for images in batch_images]
+        original_image_sizes = [[get_image_size(image, channel_dim=input_data_format) for image in images] for images in batch_images]
 
         if do_resize:
             batch_images = [
                 [self.resize(image, size=size, input_data_format=input_data_format) for image in images]
                 for images in batch_images
             ]
 
-        image_sizes = [get_image_size(images[0], channel_dim=input_data_format) for images in batch_images]
-        image_unpadded_heights = [[image_size[0]] for image_size in image_sizes]
-        image_unpadded_widths = [[image_size[1]] for image_size in image_sizes]
+        batch_image_sizes = [[get_image_size(image, channel_dim=input_data_format) for image in images] for images in batch_images]
```

Comment:
I believe the issue is not with the PR itself, but when I execute the code from the PR, I encounter a problem with images being excessively padded.

<img width="1139" alt="image" src="https://github.com/huggingface/transformers/assets/20262536/3dd319ad-90a9-424f-a8cc-3c80f02478e1">

Like this